api/provider/aws/test_aws/task_list_test.go

```diff
@@ -0,0 +1,205 @@
+		{
+			EntityID:   "tsk_id1",
+			EntityType: "task",
+			Key:        "name",
+			Value:      "tsk_name1",
+		},
+		{
+			EntityID:   "tsk_id1",
+			EntityType: "task",
+			Key:        "environment_id",
+			Value:      "env_id1",
+		},
+		{
+			EntityID:   "tsk_id1",
+			EntityType: "task",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task/arn1",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "name",
+			Value:      "tsk_name2",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "environment_id",
+			Value:      "env_id1",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task/arn2",
+		},
+		{
+			EntityID:   "tsk_id3",
+			EntityType: "task",
+			Key:        "name",
+			Value:      "tsk_name3",
+		},
+		{
+			EntityID:   "tsk_id3",
+			EntityType: "task",
+			Key:        "environment_id",
+			Value:      "env_id2",
+		},
+		{
+			EntityID:   "tsk_id3",
+			EntityType: "task",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task/arn3",
+		},
+		{
+			EntityID:   "tsk_id4",
+			EntityType: "task",
+			Key:        "name",
+			Value:      "tsk_name4",
+		},
+		{
+			EntityID:   "tsk_id4",
+			EntityType: "task",
+			Key:        "environment_id",
+			Value:      "env_id2",
+		},
+		{
+			EntityID:   "tsk_id4",
+			EntityType: "task",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task/arn4",
+		},
+		{
+			EntityID:   "env_id1",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "env_name1",
+		},
+		{
+			EntityID:   "env_id2",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "env_name2",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	listClusterPagesFN := func(input *ecs.ListClustersInput, fn func(output *ecs.ListClustersOutput, lastPage bool) bool) error {
+		clusterARNs := []*string{
+			aws.String("arn:aws:ecs:region:012345678910:cluster/l0-test-env_id1"),
+			aws.String("arn:aws:ecs:region:012345678910:cluster/l0-test-env_id2"),
+		}
+
+		output := &ecs.ListClustersOutput{}
+		output.SetClusterArns(clusterARNs)
+
+		fn(output, true)
+
+		return nil
+	}
+
+	mockAWS.ECS.EXPECT().
+		ListClustersPages(&ecs.ListClustersInput{}, gomock.Any()).
+		Do(listClusterPagesFN).
+		Return(nil)
+
+	newListTaskPagesFN := func(taskARN int) func(input *ecs.ListTasksInput, fn func(output *ecs.ListTasksOutput, lastPage bool) bool) error {
+		listTaskPagesFN := func(input *ecs.ListTasksInput, fn func(output *ecs.ListTasksOutput, lastPage bool) bool) error {
+			arn := []*string{aws.String(fmt.Sprintf("arn:aws:ecs:region:012345678910:task/arn%d", taskARN))}
+
+			output := &ecs.ListTasksOutput{}
+			output.SetTaskArns(arn)
+			fn(output, true)
+
+			return nil
+		}
+
+		return listTaskPagesFN
+	}
+
+	taskARNCount := 1
```

Comment:
[nit] could just used `i + j + 1`

Reply:
Not quite, since it won't hit every case, then if you try to add 1 it just complicates things and still doesn't hit every case unless you apply some conditional.
0 + 0 = 0
0 + 1 = 1
1 + 0 = 1
1 + 1 = 2

Reply:
oops, my bad!